Removed mobile carousel and added styling to match acceptance criteria

aditya000028 opened the pull request:
### Changes:
- Removed use of `MobileCarousel` component
- Removed unused variables
- Removed unused imports
- Added styling to match acceptance criteria

### Screenshots:
***************
##### Mobile
![image](https://github.com/user-attachments/assets/6f18dd6c-03f3-4c46-b36d-28e2248bec96)

##### Medium Screen
![image](https://github.com/user-attachments/assets/b1055683-0470-4cb4-b8e4-2ad94e90ad0a)


##### Large Screen
![image](https://github.com/user-attachments/assets/02c4a34f-9b40-40ec-a1df-b850502924a8)

Closes #30

danlechambre reviewed (changes requested):
Hey Aditya, I think it would be more consistent and maintainable to use Tailwind here. I'd also prefer if the elements were set to display none ('hidden' in Tailwind) rather than being kept in the DOM. This should also allow you to remove any reference to 'carousel' in the custom classes

aditya000028 commented:
@danlechambre Good call on the Tailwind. 
Wouldn't using display: none or `hidden` still have the images in the HTML/DOM? 
Also, I'm scratching my head on how to hide/remove images based on the screen size to accomplish this task, without using `overflow`. Any ideas?

aditya000028 commented:
Hey @danlechambre , thank you for all your suggestions! I've pushed changes which I think is flexible in terms of the number of images we can display. Here is a test that I ran with more than 3 images to show you how it would look. Let me know your thoughts!
![image](https://github.com/user-attachments/assets/ff181f3f-6aa3-43e9-9553-9328e45b92ed)

danlechambre reviewed (changes requested):
Hey @aditya000028 I think you're really close here but some of the styling could be a little brittle.

One thing to keep in mind with Tailwind is that we should apply styles only at the break points we need them. For example all of our `flex` classes don't need to be introduced until the `sm` breakpoint.

Another thing to remember is that Tailwind gives us all of the usual tools of CSS, but they're only as useful as the native CSS styles they apply. A good example of this is `gap`. I _think_ gap is the reason you had to reach for the calc functions? If you apply the padding on the img itself, you don't need to account for the extra 16px. That also means we can use the simpler flex-basis classes to govern our flex layout

Here is an example implementation that applies classes at the breakpoint we need and removes the need for the brittle calc functions:
```jsx
  <div class:list={["mb-6 sm:flex sm:flex-row sm:flex-wrap sm:justify-center"]}>
    {(contentItem.images).map((image, index) => (
      <div
        class:list={[
          "sm:basis-1/2 lg:basis-1/3",
          {
            "hidden": index !== 0,
            "sm:block": index === 1,
            "lg:block": index >= 2
          }
        ]}
      >
        <img
          class="sm:p-3"
          src={getImageUrlBuilder(image.image).width(640).height(450).url()}
          alt={image.altText}
        />
      </div>
    ))}
  </div>
```

aditya000028 commented:
@danlechambre yes I agree, I think the way you have mentioned is an improved way of getting what we want. And yes, my use of the `calc` function was required because of the `gap` that I used. 
Thank you for all your suggestions. I have made the changes, and adjusted the padding/margin to better match the spacing used throughout the rest of the page. I also removed the explicit declaration of `sm:flex-row` since I believe the default direction of flex is a row.
